### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by SebastianSolisFenger**
## In this milestone, we:

- Made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
- Made sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extracted some pieces of logic into separate classes. (MODULAR)

### CODE REVIEW

File: create_rental.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    p 'Select a book from the following list by number'
+    @list_books.list_books_with_index
+    book = @book_list[gets.chomp.to_i]
+    p 'Select a person from the following list by number (not id)'
+    @list_people.list_people_with_index
+    person = @people_list[gets.chomp.to_i]
+    print 'Date: '
+    date = gets.chomp
+    [book, person, date]
+  end
```
Comment (Gambit142) [inline create_rental.rb:23]: - [x] When I try to create a rental, I get this error in the terminal:

image
 
#### There are no instance methods or instance variables called `list_books_with_index` and `list_people_with_index` in the ListBooks and ListPeople classes. Kindly recheck. 👍

File: execute.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    when 2 then @people.list_people
+    when 3 then @person.create_person
+    when 4 then @create_book.create_book
+    when 5 then @create_rental.create_rental
+    when 6 then @list_rentals.rentals_for_id
+    when 7 then @exit.exit_app
+    else p 'Please enter a number between 1 and 7'
+    end
+  end
+end
```
Comment (Gambit142) [inline execute.rb:33]: - [x] Great work applying the Single Responsibility rule here. That is great work. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SebastianSolisFenger @Trictonicmp 👋, 

Great work applying the Single responsibility rule for this milestone. It is awe-inspiring. From your code, I can figure out that there is a great understanding and collaborative spirit in the team. Keep it up. 👏 

## Highlights
- Refactored code follows the `YAGNI` principle 💯 
- Refactored code follows the `DRY` principle 💯 
- Refactored code follows the `KISS` principle 💯 
- Refactored code follows the `SOLID` principle 💯 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by arslanbisharat — APPROVED** [review]
Hi @SebastianSolisFenger & @Gambit142,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**